elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -0,0 +1,98 @@
+package com.yahoo.elide.datastores.aggregation;
+
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.datastores.aggregation.dimension.Dimension;
+import com.yahoo.elide.datastores.aggregation.dimension.TimeDimension;
+import com.yahoo.elide.datastores.aggregation.filter.visitor.FilterConstraints;
+import com.yahoo.elide.datastores.aggregation.filter.visitor.SplitFilterExpressionVisitor;
+import com.yahoo.elide.datastores.aggregation.metric.Aggregation;
+import com.yahoo.elide.datastores.aggregation.metric.Metric;
+import com.yahoo.elide.datastores.aggregation.schema.Schema;
+import com.yahoo.elide.request.Attribute;
+import com.yahoo.elide.request.EntityProjection;
+import com.yahoo.elide.request.Relationship;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class AggregationDataStoreHelper {
+
+    private Schema schema;
+    private EntityProjection entityProjection;
+
+    public AggregationDataStoreHelper(Schema schema, EntityProjection entityProjection) {
+        this.schema = schema;
+        this.entityProjection = entityProjection;
+    }
+
+    public FilterExpression getWhereFilter(FilterExpression filterExpression) {
+        FilterExpression whereFilter;
+        try {
+            whereFilter = filterExpression.accept(new SplitFilterExpressionVisitor(schema)).getWhereExpression();
+        } catch (NullPointerException npe) {
+            whereFilter = null;
+        }
+        return whereFilter;
+    }
+
+    public FilterExpression getHavingFilter(FilterExpression filterExpression) {
+        FilterExpression havingFilter;
+        try {
+            havingFilter = filterExpression.accept(new SplitFilterExpressionVisitor(schema)).getHavingExpression();
+        } catch (NullPointerException npe) {
+            havingFilter = null;
+        }
+        return havingFilter;
+    }
+
+
+    public void populateDimensionList(Set<Dimension> dimensions, Set<TimeDimension> timeDimensions) {
+        Set<String> dimensionNames = getDimensionNames();
+        Set<String> metricNames = getMetricNames(); // time dimensions are under attribute in entity projection
```

Comment:
If `dimensionNames` and `metricNames` are all from the entityProjection and we should not modify them, maybe we can make them `private final` variables for this class and assign values for them in the constructor?